### ISSUE TITLE
[unbound-method] Doesn't work on destructuring declarations

### ISSUE THREAD

**Issue by Validark** [typescript-eslint/typescript-eslint#1112]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unbound-method": "warn"
  }
}
```

```TS
class MyClass {
	public log(): void {
		print(this);
	}
}

const instance = new MyClass();

const myLog = instance.log; // properly caught
myLog();

const { log } = instance; // no problem ?

log();
```

**Expected Result** `const { log } = instance;` should get reported as bad

**Actual Result** `const { log } = instance;` is not reported

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.3.2` |
| `@typescript-eslint/parser`        | `2.3.2` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.5.1` |
| `node`                             | `12.9.1` |
| `npm`                              | `6.11.2` |

**Comment by OliverJAsh** [comment]
:+1:. Also, note that [the example in the docs](https://github.com/typescript-eslint/typescript-eslint/blob/6aa6bc738705927685e1c0dfa12ccadc3fe53d96/packages/eslint-plugin/docs/rules/unbound-method.md) suggests the destructuring variant should error, so either this is a bug or the docs are a bit misleading?

**Comment by bradzacher** [comment]
Hmm, sometimes I don't know a rule well enough to catch things like that when triaging. Thanks for flagging.

Looking closer at the tests, they are pretty clearly written incorrectly, which is probably why this isn't working

https://github.com/typescript-eslint/typescript-eslint/blob/6aa6bc738705927685e1c0dfa12ccadc3fe53d96/packages/eslint-plugin/tests/rules/unbound-method.test.ts#L217-L220
